### PR TITLE
Object manipulation 2

### PR DESCRIPTION
**Pull request by Lee-SungWon**
Screen Shot 2020-09-07 at 5 09 52 PM

I am not sure why I don't pass test.js for "pick".  My code works in console.log

### CODE REVIEW

File: object-manipulation-2/src/defaults.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      if (sourceProp[i] !== targetProp[j]) {
+        counter++;
+      }
+    }
+    if (counter === targetProp.length) {
+      target[sourceProp[i]] = source[sourceProp[i]];
+    }
+    counter = 0;
+  }
+  return target;
```
Comment (thebearingedge) [inline object-manipulation-2/src/defaults.js]: This solution can be simplified quite a bit. Try:
1. looping through source keys
1. if the target is undefined at a source key
1. assign source at key to target at key

File: object-manipulation-2/src/invert.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  const sourceProp = [];
+
+  const sourceValue = [];
+
+  const newObj = {};
+
+  for (const property in obj) {
+    sourceProp.push(property);
+  }
```
Comment (thebearingedge) [inline object-manipulation-2/src/invert.js]: You can do all of the modification to `newObj` here. No arrays required.

File: object-manipulation-2/src/pick.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  const obj = {};
+
+  for (const property in source) {
+    for (let i = 0; i < keys.length; i++) {
+      if (property === keys[i]) {
+        obj[keys[i]] = source[keys[i]];
+      }
+    }
+  }
```
Comment (thebearingedge) [inline object-manipulation-2/src/pick.js:12]: This is pretty good, see if you can apply similar strategies to some of your other solutions.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: object-manipulation-2/src/defaults.js (hunk 1 above)
Comment (Lee-SungWon) [reply object-manipulation-2/src/defaults.js]: revised!

File: object-manipulation-2/src/invert.js (hunk 2 above)
Comment (Lee-SungWon) [reply object-manipulation-2/src/invert.js]: revised!

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Hi Chris, I don't see your revisions here, maybe the commit wasn't pushed yet?

**Comment by Lee-SungWon** [comment]
Hey Tim, I tried pushing the commits again, and the revised portions seem to be showing now
I will delete the commented out sections from now on

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**